connectors/src/connectors/microsoft/index.ts

```diff
@@ -0,0 +1,175 @@
+import type {
+  ConnectorPermission,
+  ModelId,
+  NangoConnectionId,
+} from "@dust-tt/types";
+import { assertNever } from "@dust-tt/types";
+
+import type { ConnectorPermissionRetriever } from "@connectors/connectors/interface";
+import { launchMicrosoftSharepointFullSyncWorkflow } from "@connectors/connectors/microsoft/sharepoint/temporal/client";
+import { launchMicrosoftTeamsFullSyncWorkflow } from "@connectors/connectors/microsoft/teams/temporal/client";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+export type MicrosoftConnectorType = "microsoft_sharepoint" | "microsoft_teams";
```

Comment:
Hesistating here between having a bunch of xxxMicrosoftConnectors funcitons with a switch inside depending on the connector type, or directly having each function for the connector xxxMicro

I understand the DRYing purpose, but in that case not sure we would dry a lot, while separating the functions in their respective connectors could be cleaner / more readable? WDYT?